apache-spark-master/runtime/Dockerfile

```diff
@@ -6,17 +6,14 @@ RUN apt-get update && \
   apt-get install -y \
     python \
     libnss3 \
-    openjdk-8-jre-headless \
-    curl
-
-# pull down spark jar
-RUN mkdir /opt/apache-spark && \
-  curl http://apache.osuosl.org/spark/spark-1.5.2/spark-1.5.2-bin-hadoop2.6.tgz \
-  | tar --strip-components=1 -xzC /opt/apache-spark
+    openjdk-8-jdk
 
 # add spark to path
+COPY apache-spark /opt/apache-spark
```

Comment:
I think this more or less defeats the purpose of having separate containers because it copies the entire source tree including all intermediate build artifacts.

Can we try building a distribution tarball with 

https://spark.apache.org/docs/1.5.2/building-spark.html#building-a-runnable-distribution

and untarring it in the runtime container?